src/utils/fetchWrap.ts

```
@@ -0,0 +1,26 @@
+import * as Sentry from '@sentry/browser';
+
+const sentryErrorHandler = (error: string, errorID: string) => {
+  Sentry.withScope(scope => {
+    scope.setTag('Error ID', errorID);
+    Sentry.captureException(error);
+  });
+  return error;
+};
+
+const fetchHandler = (response: Response, errorID: string) => {
+  if ((!response.ok && response.status !== 300) || 400 || 500) {
+    throw TypeError(`Front End Error ID: ${errorID}`);
+  }
+  return response;
+};
+
+export const fetchWrap = async (request: Request, errorID: string): Promise<unknown> => {
+  try {
+    const response = await fetch(request);
+    fetchHandler(response, errorID);
+  } catch (error) {
+    return sentryErrorHandler(error, errorID);
```

Instead of returning the error, we can `throw` it (same as rejecting the promise). That way places using fetchWrap can have a `try/catch` or `.then/.catch` and handle their additional processing that's specific to their request. This should also let you change the return type to `Promise<Response>`

There was a thing I proposed before about doing a `finally` block and enabling places calling the fetchWrapper to implement their own logic/callback/promise for the sentry handler. But then, that's another question.

Is this fetch wrapper to specifically catch errors like 300s, 400s, 500s, etc, or is it also to report errors that spawn from bad code. I think the error boundary would be for catching that bad code, right? So then this would be adequate without the `finally` thing. Is that right?